cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/ReadThroughBookCache.java

```
@@ -51,17 +51,20 @@ public ReadThroughBookCache(CradleOperators ops,
     }
 
     public BookInfo getBook (BookId bookId) throws CradleStorageException {
-        if (!books.containsKey(bookId)) {
-            logger.info("Book '{}' is absent in cache, trying to get it from DB", bookId.getName());
-            try {
-                books.put(bookId, loadBook(bookId));
-            } catch (CradleStorageException e) {
-                logger.error("Could not load book named {}: {}", bookId.getName(), e);
-                throw e;
-            }
+        BookInfo bookInfo = books.computeIfAbsent(bookId, bookId1 -> {
+                try {
+                    return loadBook(bookId1);
+                } catch (CradleStorageException e) {
+                    logger.error("Could not load book named {}: {}", bookId1.getName(), e);
+                    return null;
```

instead of returning **null** exception should be propagated